lib/rules/no-shadow.js

```diff
@@ -284,6 +484,30 @@ module.exports = {
 			const inner = getNameRange(variable);
 			const outer = getNameRange(scopeVar);
 
+			if (!inner || !outer || inner[1] >= outer[0]) {
+				return false;
+			}
+
+			if (!outerDef) {
+				return true;
+			}
+
+			if (hoist === "functions") {
+				return !FUNCTIONS_HOISTED_NODES.has(outerDef.node.type);
```

Comment:
[Testing] I can remove both this line and the next `FUNCTIONS_HOISTED_NODES` one without failing any tests. Which feels like a test gap. 🤔

Reply:
I think this is unnecessary code; `hoist === "functions"` is already handled by the existing rule.